lib/topic_query.rb

```diff
@@ -205,6 +215,15 @@ def list_related_for(topic, pm_params: nil)
 
   # Return a list of suggested topics for a topic
   def list_suggested_for(topic, pm_params: nil)
+    if @@custom_list_suggested_for&.present?
+      @@custom_list_suggested_for.each do |key, filter|
+        suggested = filter.call(topic, pm_params, self)
+        if suggested && !suggested[:result].blank?
+          return create_list(:suggested, suggested[:params], suggested[:result])
```

Comment:
Should we be creating our own list here vs leaning on [SuggestedTopicsBuilder#add_results](https://github.com/discourse/discourse/blob/4350a903ec5370ef7ec0aca7cafc553fc106c22b/lib/suggested_topics_builder.rb#L13-L41) which respects the `limit_suggested_to_category` site setting and excludes category topics?

Reply:
Thanks for the feedback, adjusted accordingly on 6e00c7a